MoneySplitter.Services/API/TransactionsApiService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using MoneySplitter.Infrastructure;
+using MoneySplitter.Models;
+using MoneySplitter.Services.DataModels;
+using MoneySplitter.Services.Inerfaces;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace MoneySplitter.Services.Api
+{
+    public class TransactionsApiService:ITransactionsApiService
+    {
+        private readonly IApiUrlBuilder _apiUrlBuilder;
+        private readonly IQueryApiService _queryApiService;
+        private readonly IMapper _mapper;
+        private readonly IExecutor _executor;
+
+        public TransactionsApiService(IApiUrlBuilder apiUrlBuilder, IQueryApiService queryApiService, IMapper mapper, IExecutor executor)
+        {
+            _apiUrlBuilder = apiUrlBuilder;
+            _queryApiService = queryApiService;
+            _mapper = mapper;
+            _executor = executor;
+        }
+
+        public async Task<ExecutionResult<IEnumerable<TransactionModel>>> GetAllUserTransactions()
+        {
+            var result = new ExecutionResult<IEnumerable<TransactionModel>>
+            {
+                IsSuccess = false
+            };
+
+            var apiUrlGetAllFriends = _apiUrlBuilder.GetAllUserTransactions();
+
+            IEnumerable<TransactionData> userTransactionsData = null;
+
+            await _executor.ExecuteWithRetryAsync((System.Func<Task>)(async () =>
+            {
+                userTransactionsData = await _queryApiService.GetAsync<IEnumerable<TransactionData>>( (System.Uri)apiUrlGetAllFriends);
```

Comment:
Why you casting to (System.Uri)? Is this property Uri already?

Reply:
fixed